godot-macros/src/lib.rs

```diff
@@ -259,11 +259,128 @@ mod util;
 /// }
 /// ```
 ///
+/// For exporting properties to the editor, you can use the `#[export]` attribute:
+///
+/// ```
+/// use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// struct MyStruct {
+///     #[export]
+///     my_field: i64,
+/// }
+///
+/// #[godot_api]
+/// impl MyStruct {}
+/// ```
+///
+/// If you dont also include a `#[var]` attribute, then a default one will be generated.
+/// `#[export]` also supports all of GDScript's annotations, in a slightly different format. The format is
+/// translated from an annotation by following these four rules:
+///
+/// - `@export` becomes `#[export]`
+/// - `@export_{name}` becomes `#[export(name)]`
+/// - `@export_{name}(elem1, ...)` becomes `#[export(name = (elem1, ...))]`
+/// - `@export_{flags/enum/enum_suggestion}("elem1", "elem2:key2", ...)`
+///   becomes
+///   `#[export(flags/enum/enum_suggestion = (elem1, elem2 = key2, ...))]`
+///
+///
+/// As an example of some different export attributes:
+///
+/// ```
+/// use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// struct MyStruct {
+///     // @export
+///     #[export]
+///     float: f64,
+///     
+///     // @export_range(0.0, 10.0, or_greater)
+///     #[export(range = (0.0, 10.0, or_greater))]
+///     range_f64: f64,
+///
+///     // @export_file
+///     #[export(file)]
+///     file: GodotString,
+///
+///     // @export_file("*.gd")
+///     #[export(file = "*.gd")]
+///     gdscript_file: GodotString,
+///
+///     // @export_flags_3d_physics
+///     #[export(flags_3d_physics)]
+///     physics: u32,
+///
+///     // @export_exp_easing
+///     #[export(exp_easing)]
+///     ease: f64,
+///
+///     // @export_enum("One", "Two", "Ten:10", "Twelve:12", "Thirteen")
+///     #[export(enum = (One, Two, Ten = 10, Twelve = 12, Thirteen))]
+///     exported_enum: i64,
+///
+///     // @export_flags("A:1", "B:2", "AB":3)
```

Comment:
```suggestion
///     // @export_flags("A:1", "B:2", "AB:3")
```
Just noticed this is wrong but dont feel like fixing it now

Reply:
No worries, we can revisit this doc over time and iron out the little things... Once people start using it, there is likely other stuff we'd need to add or amend.